src/olympia/abuse/migrations/0028_cinderdecision_backfill.py

```diff
@@ -0,0 +1,67 @@
+# Generated by Django 4.2.10 on 2024-03-14 18:30
+
+from django.db import migrations
+
+def backfill_cinder_decisions_from_cinder_jobs(apps, schema_editor):
+    CinderJob = apps.get_model('abuse', 'CinderJob')
+    CinderDecision = apps.get_model('abuse', 'CinderDecision')
+    Addon = apps.get_model('addons', 'Addon')
+    cinder_jobs = {}
+    decisions_to_create = []
+    policies_per_job = {}
+    qs = CinderJob.objects.exclude(decision_action=0).exclude(decision_cinder_id='').prefetch_related('policies')
+    # generate the list of CinderDecision instances to create
+    for cinder_job in qs:
+        abuse_report = cinder_job.abusereport_set.first()
+        if not abuse_report:
+            continue
+        if abuse_report.guid:
+            addon = cinder_job.target_addon or Addon.unfiltered.filter(guid=abuse_report.guid).first()
+            if not addon:
+                continue
+        else:
+            addon = None
+        decision = CinderDecision(
+            action=cinder_job.decision_action,
+            cinder_id=cinder_job.decision_cinder_id,
+            date=cinder_job.decision_date,
+            notes=cinder_job.decision_notes,
+            appeal_job=cinder_job.appeal_job,
+            addon=addon,
+            user=abuse_report.user,
+            rating=abuse_report.rating,
+            collection=abuse_report.collection,
+        )
+        decisions_to_create.append(decision)
+        # also track the new m2m relationships we'll need the policies, and the back links to the job
+        policies_per_job[cinder_job.decision_cinder_id] = list(cinder_job.policies.all())
+        cinder_jobs[cinder_job.decision_cinder_id] = cinder_job
+
+    # Bulk create the CinderDecisions
+    CinderDecision.objects.bulk_create(decisions_to_create)
+
+    m2m_policies_to_create = []
+    # Loop through the CinderDecisions, to generate the m2m list and to set cinderjob.decision
+    for decision in CinderDecision.objects.all():
```

Comment:
Having to do a query to get the decisions was a little annoying, but it appears mysql doesn't return auto generated ids so (unlike, say, postgres) `bulk_create` doesn't return objects that have been saved, so they can't be used for foreign keys.